zone.go

```diff
@@ -865,6 +896,23 @@ func (api *API) UpdateZoneSSLSettings(ctx context.Context, zoneID string, sslVal
 	return r.Result, nil
 }
 
+// UpdateZoneAutomaticSSLSettingMode update information about Automatic SSL mode setting to the specified zone.
+//
+// API reference: https://api.cloudflare.com/[#-add-url-slug-here]
+func (api *API) UpdateZoneAutomaticSSLSettingMode(ctx context.Context, zoneID, mode string) (ZoneSSLSetting, error) {
```

Comment:
we already have a method for fetching/updating zone settings (https://github.com/cloudflare/cloudflare-go/blob/master/zone.go#L958). is there a reason we cannot reuse that here?

Reply:
Correct, however, there is a slight difference in the response we return for this specific setting. As it returns a field called `next_scheduled_scan`: https://github.com/cloudflare/cloudflare-go/pull/2855/files#diff-ceb3700215079abfa8dfc51b35af780ed6292f4747db0d4710c6c72de18a0a66R178

Also I think the return type here should reference the `ZoneAutomaticSSLModeSetting` struct vs `ZoneSSLSetting`. Will fix that typo.

Reply:
you could always put it in a pointer with `omitempty` which will conditionally set it when it is provided.

Reply:
in the generated version of the libraries, these endpoints are flattened into a single struct so that would make sense to reflect here too if possible.

Reply:
updated to use the existing Get/UpdateZoneSettings construct and added the field as part of the ZoneSetting struct with omitempty markup